File: Euclid/Messages/Incoming/GOAWAY.cs
using Euclid.Game;
using Euclid.Network.Streams;

namespace Euclid.Messages.Incoming
{
    class GOAWAY : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            if (player.RoomEntity.Room == null)
            {
                return;
            }

            if (!player.RoomEntity.WalkingAllowed)
            {
                return;
            }

            player.RoomEntity.BeingKicked = true;
            player.Disconnect();
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/GOAWAY.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;

public class GOAWAY implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (player.getRoomEntity().getRoom() == null) {
            return;
        }

        if (!player.getRoomEntity().isWalkingAllowed()) {
            return;
        }

        player.getRoomEntity().setBeingKicked(true);
        player.disconnect();
    }
}
